pyfifotax/report_data.py

```diff
@@ -0,0 +1,535 @@
+                tmp = self.held_shares[event.symbol].pop(
+                    event.quantity, event.sell_price, event.date
+                )
+                self.sold_shares[event.symbol].extend(tmp)
+
+                self.held_forex[event.currency].push(event.received_forex)
+
+                if event.paid_fees is not None:
+                    self.misc["Fees"].append(event.paid_fees)
+
+                self.awv_z10_events.append(
+                    awv.AWVEntryZ10Sale(
+                        date=event.date,
+                        symbol=event.symbol,
+                        currency=event.currency,
+                        quantity=event.quantity,
+                        value=event.quantity * event.sell_price,
+                    )
+                )
+
+            elif isinstance(event, MoneyDepositEvent):
+                new_forex = FIFOForex(
+                    currency=event.currency,
+                    quantity=(
+                        event.amount - event.fees.amount
+                        if event.fees is not None
+                        else event.amount
+                    ),
+                    buy_date=event.buy_date,
+                    source=f"Deposit of Money",
+                )
+                self.held_forex[event.currency].push(new_forex)
+                if event.fees is not None:
+                    self.misc["Fees"].append(event.fees)
+
+            elif isinstance(event, MoneyWithdrawalEvent):
+                if event.fees is not None:
+                    self.misc["Fees"].append(event.fees)
+                try:
+                    tmp = self.held_forex[event.currency].pop(
+                        event.amount,
+                        to_decimal(1),
+                        event.date,
+                    )
+                except ValueError:
+                    raise ValueError(
+                        f"Cannot withdraw {event.amount} {event.currency}, not enough owned."
+                    )
+                self.withdrawn_forex[event.currency].extend(tmp)
+
+            elif isinstance(event, CurrencyConversionEventFromEURToForex):
+                # "buy" forex (note: EUR used up not tracked)
+                new_forex = FIFOForex(
+                    currency=event.target_currency,
+                    quantity=event.foreign_amount,
+                    buy_date=event.date,
+                    source=f"Currency Conversion EUR to {event.target_currency}",
+                )
+                self.held_forex[event.target_currency].push(new_forex)
+                if event.source_fees is not None:
+                    self.misc["Fees"].append(event.source_fees)
+
+            elif isinstance(event, CurrencyConversionEventFromForexToEUR):
+                # "sell" forex (note: EUR received not tracked)
+                tmp = self.held_forex[event.source_currency].pop(
+                    event.foreign_amount,
```

Comment:
There're `source` and `target` currencies, which are clear. However, `foreign` seems to be a third option which requires a bit of mental gymnastics to bind it to one of the two. I remember that it led me astray once.

Could you please change the column name from `foreign_amount` to `target_amount`? I think it'll be much more obvious what number one has to write there.